app/services/csv/lettings_log_csv_service.rb

```diff
@@ -5,10 +5,20 @@ def initialize(user:, export_type:, year:)
       @export_type = export_type
       @year = year
       @attributes = lettings_log_attributes
+      @definitions = lettings_log_definitions
     end
 
     def prepare_csv(logs)
       CSV.generate(headers: true) do |csv|
+        if @year >= 2023
+          csv << @attributes.map do |attribute|
+            definition = @definitions.find { |record| record.variable == attribute.downcase }
```

Comment:
would it make sense for @definitions to be something like `{created_by: "definition", id: "definition"}` ?
Then you could access them as `@definitions[attribute.downcase]` here

Reply:
I still want access to the record itself to do the next step and update!, otherwise yes that would work too.